### ISSUE TITLE
Unpacking fails dangerous-subprocess-use-audit

### ISSUE THREAD

**Issue by CandiedCode** [returntocorp/sgrep-rules#3074]
**Describe the bug**
A clear and concise description of what the bug is and for which rule. Is the rule producing a false positive? Is it incorrect or missing something?
The rule doesn't recognize unpacking, thus produces a false positive.

**To Reproduce**
Sample code to reproduce this behavior.
```python
# ok:dangerous-subprocess-use-audit
subprocess.call(["echo", "a", ";", "rm", "-rf", "/"])

echo_cmd = ["echo", "a", ";"]
# should be okay but fails dangerous-subprocess-use-audit
subprocess.call([*echo_cmd, "rm", "-rf", "/"])
```

**Expected behavior**

The unpacked example should pass.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**
